### PR TITLE
Added map marker clustering

### PR DESCRIPTION
**Pull request by Waltzerizm**
- Now the markers cluster when there is multiple markers in one place
- Added some data to the location database for marker clustering demonstration
- Installed Xamarin.Forms.GoogleMaps.Clustering NuGet package

### CODE REVIEW

File: Encountify/Views/MapPage.xaml [hunk 1]
```diff
@@ -15,11 +15,11 @@
...
             </Grid.RowDefinitions>
 
-            <maps:Map 
-                x:Name="map" 
-                VerticalOptions="FillAndExpand"
-                MyLocationEnabled="True"/>
-            
+            <clustering:ClusteredMap x:Name="map"
+                                 HorizontalOptions="FillAndExpand"
+                                 VerticalOptions="FillAndExpand"
```
Comment (Mdominykas) [inline Encountify/Views/MapPage.xaml:20]: Seems that there are 17 extra spaces

**Review by Mdominykas — CHANGES_REQUESTED** [review]
It works well, however I noticed that when you only have 2 locations and LoadDummyData will try to add more, it will raise and exception and not load anything, since location1 already exists in the database.
For that I believe MockDataStore needs to be nearly completely rewrote (I have already investigated it, so it would be quite easier for me, thus I will try to change it tommorow).
for now to avoid exceptions I suggest addiding location1, location2, ..., location8 to list and inserting to database only those that exceed db.Table<Location>.Count()

File: Encountify/Views/MapPage.xaml (hunk 1 above)
Comment (Waltzerizm) [reply Encountify/Views/MapPage.xaml]: Didn't noticed, thank you! Fixed

**Comment by Waltzerizm** [comment]
Reverted the changed on the LoadDummyData() method since Dominykas work on it. Clustering option is still there, so if you want to see it make sure you have enough markers in one spot added!

File: Encountify/Services/MockDataStore.cs [hunk 2]
```diff
@@ -61,6 +61,7 @@ public void LoadDummyData(SQLiteConnection db)
             {
                 var location1 = new Location() { Name = "Vilniaus katedra", CoordX = 54.685849042698216, CoordY = 25.287750880122083 };
                 var location2 = new Location() { Name = "Gedimino bokštas", CoordX = 54.68667445192699, CoordY = 25.29056883194689 };
+
```
Comment (Mdominykas) [inline Encountify/Services/MockDataStore.cs:64]: If you remove this new line I will likely not have to solve merge conflict :)

**Review by Mdominykas — APPROVED** [review]
It works well for me

**Review by MigleMic — APPROVED** [review]
Just checked it out, works great :)

**Pull request merged**